dz_s9/zadacha_1_DZ.py

```diff
@@ -0,0 +1,66 @@
+# Напишите следующие функции:
+# Нахождение корней квадратного уравнения
+# Генерация csv файла с тремя случайными числами в каждой строке. 100-1000 строк.
+# Декоратор, запускающий функцию нахождения корней квадратного уравнения с каждой тройкой чисел из csv файла.
+# Декоратор, сохраняющий переданные параметры и результаты работы функции в json файл.
+import json
+import os.path
+from random import randint
+import csv
+
+__all__ = ['create_csv', 'quadratic_equation', 'deco_save_param_result']
+
+
+def deco_save_param_result(func):
+    def wrapper(a, b, c):
+        filename = 'results.json'
+        if os.path.exists(filename):
+            with open(filename, 'r', encoding='utf-8') as file:
+                list_file = json.load(file)
+        else:
+            list_file = []
+        result = func(a, b, c)
+        list_file.append({' '.join(map(str, (a, b, c))): result})
+        with open(filename, 'w', encoding='utf-8') as file:
+            json.dump(list_file, file, indent=1)
+        return result
+
+    return wrapper
+
+
+def deco_quadr_eq_from_csv(func):
+    def wrapper():
+        with open('numbers.csv', 'r', newline='', encoding='utf-8') as file:
+            numbers = csv.reader(file)
+            for line in numbers:
+                result = func(*(map(int, line)))
+        return result
```

Comment:
В задании указано, что декоратор должен только запускать функцию в цикле. Про возврат речь не шла. Можно смело это убрать, функциональность не поменяется.
```suggestion
                func(*(map(int, line)))
```